src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -250,22 +252,28 @@ private static FactoryContext CreateFactoryContext(RequestDelegateFactoryOptions
         return HandleRequestBodyAndCompileRequestDelegate(responseWritingMethodCall, factoryContext);
     }
 
-    private static RouteHandlerFilterDelegate CreateFilterPipeline(MethodInfo methodInfo, Expression? target, FactoryContext factoryContext)
+    private static RouteHandlerFilterDelegate CreateFilterPipeline(MethodInfo methodInfo, Expression? targetExpression, FactoryContext factoryContext, Expression<Func<HttpContext, object?>>? targetFactory)
     {
         Debug.Assert(factoryContext.Filters is not null);
         // httpContext.Response.StatusCode >= 400
         // ? Task.CompletedTask
-        // : handler((string)context.Parameters[0], (int)context.Parameters[1])
+        // : {
+        //  target = targetFactory(httpContext);
+        //  handler((string)context.Parameters[0], (int)context.Parameters[1]);
+        // }
         var filteredInvocation = Expression.Lambda<RouteHandlerFilterDelegate>(
             Expression.Condition(
                 Expression.GreaterThanOrEqual(FilterContextHttpContextStatusCodeExpr, Expression.Constant(400)),
                 CompletedValueTaskExpr,
                 Expression.Block(
-                    new[] { TargetExpr },
+                    new[] { TargetExpr, HttpContextExpr },
```

Comment:
The HttpContextExpr here worries me. Looks like it might define a local which wouldn't be good. I think we should instead be using FilterContextHttpContextExpr directly in the Expression.Invoke below

Reply:
Good call!